doc/source/skips/4-transition-to-v2.rst

```diff
@@ -0,0 +1,249 @@
+
+Detailed description
+--------------------
+
+It is beyond the scope of this document to list all of the proposed API changes
+for skimage2, many of which have yet to be decided upon. Indeed, the
+scope and ambition of the 2.0 transition could grow if this SKIP is accepted.
+This SKIP instead proposes a mechanism for managing the transition without
+breaking users' code. A meta-issue tracking the proposed changes can be found
+on GitHub, scikit-image/scikit-image#5439 [7]_. Some examples are briefly
+included below for illustrative purposes:
+
+- Stop rescaling input arrays when the dtype must be coerced to float.
+- Stop swapping coordinate axis order in different contexts, such as drawing or
+  warping.
+- Allow automatic return of non-NumPy types, so long as they are coercible to
+  NumPy with ``numpy.asarray``.
+- Harmonizing similar parameters in different functions to have the same name;
+  for example, we currently have ``random_seed``, ``random_state``, ``seed``,
+  or ``sample_seed`` in different functions, all to mean the same thing.
+- Changing ``measure.regionprops`` to return a dictionary instead of a list.
+- Combine functions that have the same purpose, such as ``watershed``,
+  ``slic``, or ``felzenschwalb``, into a common namespace. This would make it
+  easier for new users to find out which functions they should try out for a
+  specific task. It would also help the community grow around common APIs,
+  where now scikit-image APIs are essentially unique for each function.
+
+To make this transition with a minimum amount of user disruption, this SKIP
+proposes releasing a new library, skimage2, that would replace the existing
+library, *but only if users explicitly opt-in*. Additionally, by releasing a
+new library, users could depend *both* on scikit-image (1.0) and on skimage2,
+allowing users to migrate their code progressively.
+
+Related Work
+------------
+
+`pandas` released 1.0.0 in January 2020, including many backwards-incompatible
+API changes [3]_. `scipy` released version 1.0 in 2017, but, given its stage of
+maturity and position at the base of the scientific Python ecosystem, opted not
+to make major breaking changes [4]_. However, SciPy has adopted a policy of
+adding upper-bounds on dependencies [5]_, acknowledging that the ecosystem as a
+whole makes backwards incompatible changes on a 2 version deprecation cycle.
+
+Several libraries have successfully migrated their user community to a new
+namespace with a version number on it, such as OpenCV (imported as ``cv2``) and
+BeautifulSoup (imported as ``bs4``), Jinja (``jinja2``) and psycopg (currently
+imported as ``psycopg2``). Further afield, R's ggplot is used as ``ggplot2``.
+
+Implementation
+--------------
+
+The details of the proposal are as follows:
+
+- scikit-image 0.19 will be followed by scikit-image 1.0. Every deprecation
+  message will be removed from 1.0, and the API will be considered the
+  scikit-image 1.0 API.
```

Comment:
To clarify by "every deprecation message will be removed", you mean that just the warnings will be removed, but the actual deprecations will not be completed?. For instance, rather than removing `multichannel` in favor of `channel_axis`, both would be left in the 1.0 API? (to avoid having a shorter than usual deprecation cycle for deprecations that were new to v0.19).

Reply:
Yeah it's ambiguous, and I agree with your assessment: let's make only the deprecations slated for 0.20, and remove other warning messages, even if it means a duplicate API (as in the case of multichannel vs channel_axis).